packages/clerk-js/src/ui/components/SessionTasks/index.tsx

```diff
@@ -71,9 +71,14 @@ export const SessionTasks = withCardStateProvider(() => {
   const signInContext = useContext(SignInContext);
   const signUpContext = useContext(SignUpContext);
   const currentTaskContainer = useRef<HTMLDivElement>(null);
+  const { queryParams } = useRouter();
 
   const redirectUrlComplete =
-    signInContext?.afterSignInUrl ?? signUpContext?.afterSignUpUrl ?? clerk?.buildAfterSignInUrl();
+    signInContext?.afterSignInUrl ??
```

Comment:
`SignInContext` and `SignUpContext` seems to be returning `afterSignInUrl/SignUpUrl` as undefined in those cases

We should get the redirect URL from either `SignIn` or `SignUp` as a prop, instead of building it here from `clerk`, since their React context already has the logic to extract from query params: 
https://github.com/clerk/javascript/blob/2fa2a4193760dbea6dbd0bbe1315470b0c07488b/packages/clerk-js/src/ui/contexts/components/SignUp.ts#L63-L78